tools/libfrencutils/mosaic_util.c

```diff
@@ -521,12 +521,12 @@ double spherical_angle(const double *v1, const double *v2, const double *v3)
   qz = v1[0]*v3[1] - v1[1]*v3[0];
 
   ddd = (px*px+py*py+pz*pz)*(qx*qx+qy*qy+qz*qz);
-  if ( ddd <= 0.0 ) 
+  if ( ddd <= 0.0 )
     angle = 0. ;
   else {
     ddd = (px*qx+py*qy+pz*qz) / sqrt(ddd);
-    if( fabs(ddd-1) < EPSLN30 ) ddd = 1;
-    if( fabs(ddd+1) < EPSLN30 ) ddd = -1;
+    if( fabsl(ddd-1) < EPSLN30 ) ddd = 1;
+    if( fabsl(ddd+1) < EPSLN30 ) ddd = -1;
```

Comment:
fabsl is valid only when NO_QUAD_PRECISION is false.  Otherwise, we would need to use fabs.  I recommend adding a macro in the #ifdef/#else/#endif at 508ff to choose the correct version of the 'abs' function.

Reply:
@bensonr should quad precision be enables on all systems that support `long double`, or should it be something the user turns on/off with an option to `configure`?

Reply:
Looking over the remaining code, `long double` is used in many places outside of the `NO_QUAD_PRECISION`.  `NO_QUAD_PRECISION` is only used in `mosaic_util.c`  and only in the `spherical_angle` function.  Thus, I don't know if it is worth leaving in this this option, and leave it to the compiler to know if `long double` has more precision than `double`.  From what I can tell in my research, all current C compilers support `long double`, but do not guarantee a larger range or precision over `double`.

I can add a test to `configure.ac` to see if `long double` has more precision than `double` and add a flag `--with-quad-precision` (default to yes) to issue a warning/error if quad-precision does not have any addition precision.  Thoughts?

And as an FYI, I found that the C preprocessor has the ability to do compile time substitution of the correct symbol based on the lvalue.  Thus, the following can choose between the `long double` and `double` form of `abs`:

```
#define fabs(X) _Generic((X), \
     long double: fabsl, \
         default: fabs)(X)
```

Then in the code, you can do `ans = fabs(var)`, and the preprocessor will use the correct form of `fabs` based on the type.